lib/cea/cea608_memory.js

```diff
@@ -0,0 +1,513 @@
+/*! @license
+ * Shaka Player
+ * Copyright 2016 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+goog.provide('shaka.cea.Cea608Memory');
+goog.provide('shaka.cea.Cea608Char');
+
+/**
+ * CEA-608 captions memory/buffer.
+ */
+shaka.cea.Cea608Memory = class {
+  /**
+   * @param {!shaka.cea.CeaDecoder} decoder CEA-608 decoder.
+   * @param {!number} fieldNum Field number.
+   * @param {!number} chanNum Channel number.
+   * @param {!boolean} textmode Indicates whether text mode is on/off.
+   */
+  constructor(decoder, fieldNum, chanNum, textmode) {
+    /**
+     * Buffer for storing decoded characters.
+     * @private @const {!Array<!Array<!shaka.cea.Cea608Char>>}
+     */
+    this.rows_ = [];
+
+    /**
+     * Current row.
+     * @private {!number}
+     */
+    this.row_ = 1;
+
+    /**
+     * Number of rows in the scroll window. Used for rollup mode.
+     * @private {!number}
+     */
+    this.scrollRows_ = 0;
+
+    /**
+     * Field number.
+     * @private {!number}
+     */
+    this.fldnum_ = fieldNum;
+
+    /**
+     * Channel number.
+     * @private {!number}
+     */
+    this.chnum_ = chanNum;
+
+    /**
+     * Text mode on/off.
+     * @private {!boolean}
+     */
+    this.textmode_ = textmode;
+
+    /**
+     * CEA-608 decoder.
+     * @private @const {!shaka.cea.CeaDecoder}
+     */
+    this.decoder_ = decoder;
+
+    /**
+     * @private {!boolean}
+     */
+    this.underline_ = false;
+
+    /**
+     * @private {!boolean}
+     */
+    this.italics_ = false;
+
+    /**
+     * @private {!string}
+     */
+    this.textColor_ = shaka.cea.Cea608Memory.DEFAULT_TXT_COLOR;
+
+    /**
+     * @private {!string}
+     */
+    this.backgroundColor_ = shaka.cea.Cea608Memory.DEFAULT_BG_COLOR;
+
+    this.reset();
+
+    // Text mode currently not emitted, so it is unused.
+    shaka.util.Functional.ignored(this.textmode_);
+  }
+
+  /**
+   * Emits a closed caption based on the state of the buffer.
+   * @param {!number} startTime Start time of the cue.
+   * @param {!number} endTime End time of the cue.
+   */
+  forceEmit(startTime, endTime) {
+    // If we emit text mode in future, prefix would be based on this.textmode.
```

Comment:
Does text mode influence the stream name somehow?  Sorry, I don't understand this comment.